src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -135,16 +137,13 @@ class TasksResource @Inject() (
       // FIXME (gkleiman): taskKiller.kill a few lines below also checks authorization, but we need to check ALL before
       // starting to kill tasks
       affectedApps.foreach(checkAuthorization(UpdateRunSpec, _))
-      val killed = await(Future.sequence(toKill
+      val killedInstances = await(Future.sequence(toKill
         .filter { case (appId, _) => affectedApps.exists(app => app.id == appId) }
         .map {
           case (appId, instances) => taskKiller.kill(appId, _ => instances, wipe)
         })).flatten
-      ok(jsonObjString("tasks" -> killed.flatMap { instance =>
-        instance.tasksMap.valuesIterator.map { task =>
-          EnrichedTask(instance, task, Nil).toRaml
-        }
-      }))
+      val killedTasks = killedInstances.collect { case EnrichedTask.All(enrichedTasks) => enrichedTasks.map(_.toRaml) }.flatten
```

Comment:
btw. I would find something like `killedInstances.map(ki => ki.toEnrichedTasks).map(_.toRaml)` more readable...

Reply:
That does not work any more. We have to filter out all instances that have `agentInfo = None`. Since we do not type scheduled instances I cannot guarantee that the agent info is set.

However, I could do

```scala
val killedTasks = killedInstances
    .collect { case EnrichedTask.All(enrichedTasks) => enrichedTasks }
    .flatten.map(_.toRaml)
```

Reply:
oh! So it's not all but all running then? I totally missed that... should we maybe reflect that somehow somewhere?

Reply:
It is all tasks except for the scheduled ones.